### PR TITLE
feat: BTC fee UI

### PR DESCRIPTION
**Pull request by abdulhaseeb4239**
# 🔘 PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
We need to update BTC fee modal UI according to the figma

Issue Link: https://linear.app/xverseapp/issue/ENG-3284/%5Bext-%F0%9F%8C%8D%5D-edit-fees-with-presets-based-on-mempool-confirmation-times-or
Context Link (if applicable): [figma](https://www.figma.com/file/FCk8jDMsC8UEfPSWnqewLY/Xverse-%E2%80%94-Extension-V2?type=design&node-id=10826-89708&mode=design&t=XMkibAwkYfLuwYdn-0)

# 🔄 Changes

- added useBtcFee hook
- added editBtcFee component for BTC transactions
- renamed editFee to editStxFee and remove BTC logic

Impact:
- BTC transactions

# 🖼 Screenshot / 📹 Video

Screenshot 2024-01-02 at 5 48 16 PM

Screenshot 2024-01-02 at 5 48 25 PM

Screenshot 2024-01-02 at 6 05 17 PM

# ✅ Review checklist
Please ensure the following are true before merging:

- [x] Code Style is consistent with the project guidelines.
- [x] Code is readable and well-commented.
- [x] No unnecessary or debugging code has been added.
- [x] Security considerations have been taken into account.
- [x] The change has been manually tested and works as expected.
- [x] Breaking changes and their impacts have been considered and documented.
- [x] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/components/bottomModal/index.tsx [hunk 1]
```diff
@@ -11,9 +11,9 @@ const BottomModalHeaderText = styled.h1((props) => ({
 const RowContainer = styled.div((props) => ({
   display: 'flex',
   flexDirection: 'row',
-  alignItems: 'space-between',
-  margin: props.theme.spacing(12),
-  marginBottom: props.theme.spacing(10),
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  margin: props.theme.spacing(8),
```
Comment (dhriaznov) [inline src/app/components/bottomModal/index.tsx:16]:
```suggestion
  margin: props.theme.space.m,
```

File: src/app/components/transactionSetting/editBtcFee.tsx [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+}
+
+const FeeItemContainer = styled.button<FeeContainerProps>`
+  display: flex;
+  padding: 12px 16px;
+  align-items: center;
+  gap: 12px;
+  align-self: stretch;
+  border-radius: 12px;
+  border: 1px solid var(--White-850, rgba(255, 255, 255, 0.15));
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/editBtcFee.tsx:109]: Are we planning on using css variables like `--White-850` in the near future? As I can see, we don't have any yet

File: src/app/components/transactionSetting/editBtcFee.tsx [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+  isSelected: boolean;
+}
+
+const FeeItemContainer = styled.button<FeeContainerProps>`
+  display: flex;
+  padding: 12px 16px;
+  align-items: center;
+  gap: 12px;
+  align-self: stretch;
+  border-radius: 12px;
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/editBtcFee.tsx:108]:
```suggestion
  border-radius: ${(props) => props.theme.radius(2)};
```

File: src/app/components/transactionSetting/editBtcFee.tsx [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+
+interface FeeContainerProps {
+  isSelected: boolean;
+}
+
+const FeeItemContainer = styled.button<FeeContainerProps>`
+  display: flex;
+  padding: 12px 16px;
+  align-items: center;
+  gap: 12px;
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/editBtcFee.tsx:106]:
```suggestion
  gap: ${(props) => props.theme.space.s};
```

File: src/app/components/transactionSetting/editBtcFee.tsx [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+          btcRecipients[0].address,
+          ordinalTxUtxo,
+          btcAddress,
+          btcClient,
+          network.type,
+          ordinalsUtxos || [],
+          feeMode,
+          feeRateInput,
+        );
+        setFeeRateInput(selectedFeeRate!.toString());
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/editBtcFee.tsx:255]: Let's add a check for the `selectedFeeRate` to get rid of `!` symbol here to make the code safer

File: src/app/components/transactionSetting/editBtcFee.tsx [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+    if (fiatAmount) {
+      if (fiatAmount.isLessThan(0.01)) {
+        return `<${currencySymbolMap[fiatCurrency]}0.01 ${fiatCurrency}`;
+      }
+      return (
+        <NumericFormat
+          value={fiatAmount.toFixed(2).toString()}
+          displayType="text"
+          thousandSeparator
+          prefix={`~ ${currencySymbolMap[fiatCurrency]} `}
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/editBtcFee.tsx:295]:
```suggestion
          prefix={`~ ${currencySymbolMap[fiatCurrency]}`}
```

We would like to have no space between the fiat currency symbol and the amount, according to our design

File: src/app/components/transactionSetting/feeItem.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+interface FeeContainer {
+  isSelected: boolean;
+}
+
+const FeeItemContainer = styled.button<FeeContainer>`
+  display: flex;
+  padding: 12px 16px;
+  align-items: center;
+  gap: 12px;
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/feeItem.tsx:14]: Same as above

File: src/app/components/transactionSetting/feeItem.tsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      case 'medium':
+        return <CarProfile size={20} color={Theme.colors.tangerine} />;
+      case 'low':
+        return <Bicycle size={20} color={Theme.colors.tangerine} />;
+      default:
+        return <RocketLaunch size={20} color={Theme.colors.tangerine} />;
+    }
+  };
+
+  // todo: localisation
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/feeItem.tsx:85]: I think we can use the localization keys from the `SPEED_UP_TRANSACTION` object (`src/locales/en.json`) here for now

**Review by dhriaznov — COMMENTED** [review]
Good job! I didn't find anything critical, but added some suggestions to polish the code

**Comment by dhriaznov** [comment]
I think we should change the Back button style not to be the primary one
Monosnap Xverse — Extension V2 2024-01-03 10-14-58

File: src/app/components/transactionSetting/editBtcFee.tsx [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+`;
+
+const TextsRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  flex: 1;
+`;
+
+const CustomTexts = styled.div`
```
Comment (dhriaznov) [inline src/app/components/transactionSetting/editBtcFee.tsx:123]:
```suggestion
const CustomText = styled.div`
```

**Comment by dhriaznov** [comment]
It looks like the Edit fees modal currently lacks "Sats" text to the right of the fee number
293699766-c85218fe-e153-4fc2-99bd-0591d6816931
Monosnap Xverse — Extension V2 2024-01-03 10-43-42

**Comment by abdulhaseeb4239** [comment]
> I think we should change the Back button style not to be the primary one

Can you clarify? The back button is secondary here, right?

**Comment by dhriaznov** [comment]
> Can you clarify? The back button is secondary here, right?

Yes, we can add a `transparent` prop to style it as a secondary button

**Comment by dhriaznov** [comment]
Also, when I click "Back" on the Custom Fee modal, it highlights the Custom option, even though I didn't choose it and in fact I still have the Medium fee set 
Monosnap Monosnap 2024-01-04 09-49-59

**Comment by abdulhaseeb4239** [comment]
> transparent

I believe this is the expected behavior, given that the 'Custom' option was initially selected. If the intention is to modify the fee to align with the higher value, should we then select 'High' or 'Custom'? This approach might introduce confusion. wdyt?

**Comment by dhriaznov** [comment]
I believe it should have the same UX we currently have for the BTC RBF feature in the web extension @QUDBS FV

**Comment by QUDBS** [comment]
@abdulhaseeb4239 @dhriaznov if the user selects custom, doesn't enter a thing, and then comes back, the active preset is the previous preset that was selected. Custom can only be active if a value was entered

**Comment by dhriaznov** [comment]
Agreed, that's how it was implemented for the BTC RBF

**Comment by abdulhaseeb4239** [comment]
> @abdulhaseeb4239 @dhriaznov if the user selects custom, doesn't enter a thing, and then comes back, the active preset is the previous preset that was selected. Custom can only be active if a value was entered

added the change @dhriaznov

**Comment by dhriaznov** [comment]
@abdulhaseeb4239 I tested that change and it works 👍 the "Back" button still has primary styling though

also, do we want to release this next week? I saw in linear the `0.29.0` version label, which will be later because the nearest release will be `0.28.0`

**Review by dhriaznov — APPROVED** [review]
LGTM

**Comment by dhriaznov** [comment]
I just tested this PR once again (attached videos to the linear ticket), works just fine, I think we can merge it

**Pull request merged**